### PR TITLE
[IIIF-1115] Add first pass at project skeleton

### PR DESCRIPTION
**Pull request by ksclarke**
* Add project stub, including first unit/integration tests
* Add Docker image build and configuration
* Add pgsql db test container and configuration

The Docker image build has some stuff in it to build a native binary (in the container). Since this project uses the latest version of Vert.x and that has a new version of Netty, I don't have all the quirks worked out (i.e., there are some new things that are breaking my previously working configuration -- cf https://github.com/netty/netty/issues/11369). So, the build has a configurable option to build the native binary or the uber-jar. Currently, it's building an uber-jar.

### CODE REVIEW

File: src/main/java/edu/ucla/library/iiif/auth/Op.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+package edu.ucla.library.iiif.auth;
+
+/**
+ * Template API operation IDs.
+ */
+public final class Op {
+
+    /**
+     * Mints a new ARK namespace.
```
Comment (DRickard) [inline src/main/java/edu/ucla/library/iiif/auth/Op.java:10]: Some cut-and-paste javadoc here?

Comment (ksclarke) [reply src/main/java/edu/ucla/library/iiif/auth/Op.java:10]: Yep, thanks :-)

**Review by DRickard — DISMISSED** [review]

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+            <bouncer_pgsql>
+              <name>postgres:12.7-alpine</name>
+              <run>
+                <containerNamePattern>bouncer_pgsql</containerNamePattern>
+                <ports>
+                  <port>${pgsql.test.port}:5432</port>
+                </ports>
+                <volumes>
+                  <bind>
+                    <volume>${project.basedir}/src/test/resources/db/:/docker-entrypoint-initdb.d/</volume>
```
Comment (markmatney) [inline pom.xml:222]: This must be how `authzdb.sql` gets run -- cool

**Review by markmatney — CHANGES_REQUESTED** [review]
A couple stray Ctrl-C Ctrl-V's, and one question about the `bouncer` script.

File: src/main/docker/scripts/bouncer [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#! /bin/sh
+
+CONFIG="-Dvertx-config-path=/etc/bouncer/config.properties"
+LOGGER="-Dvertx.logger-delegate-factory-class-name=io.vertx.core.logging.SLF4JLogDelegateFactory"
+
+java $LOGGER -jar /opt/bouncer/bouncer.jar run edu.ucla.library.iiif.auth.verticles.MainVerticle
```
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/main/docker/scripts/bouncer:6]: For now, no. I'll go ahead and take it out. This was me thinking about how to handle configs better and this has small enough number of variables at this point that I think we can do without.

[the remark this answers is not in the document]
Comment (markmatney) [reply src/main/docker/scripts/bouncer:6]: _`12-factor`ing intensifies_

File: src/main/resources/bouncer.yaml [hunk 4]
```diff
@@ -1,17 +1,127 @@
...
+              schema:
+                type: string
+                example: The request did not contain a valid cookie
+        '500':
+          description: There was an internal server error
+          content:
+            text/plain:
+              schema:
+                type: string
+                example: The service failed to connect to its access control database
```
Comment (markmatney) [inline src/main/resources/bouncer.yaml:119]: Was this change intentional?

File: src/test/resources/db/authzdb.sql [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+GRANT SELECT ON TABLE public.items TO authz_reader;
+GRANT ALL ON TABLE public.items TO authz_writer;
+
+--
+-- Name: TABLE origins; Type: ACL; Schema: public; Owner: postgres
+--
+
+GRANT SELECT ON TABLE public.origins TO authz_reader;
+GRANT ALL ON TABLE public.origins TO authz_writer;
```
Comment (markmatney) [inline src/test/resources/db/authzdb.sql:118]: Should we have tests for the items and origins tables at this point?

File: src/main/resources/bouncer.yaml (hunk 4 above)
Comment (markmatney) [reply src/main/resources/bouncer.yaml:119]: (Just asking because the handlers are unimplemented.)

Comment (ksclarke) [reply src/main/resources/bouncer.yaml:119]: Yes, this is IIIF-1117, which is just for the OpenAPI spec (none of the handlers are implemented yet; there are different tickets for those). I was trying to keep this in a separate PR but it seems to have slipped in here.

File: src/test/resources/db/authzdb.sql (hunk 5 above)
Comment (ksclarke) [reply src/test/resources/db/authzdb.sql:118]: I just have the one database test at this point to confirm it was loaded. It's making the assumption if that one change from the SQL file was loaded that the rest are too.

I think we should leave more detailed tests to the tickets that are actually using these table structures. As we saw in our Josh meeting a few minutes ago, these may change as we get into the implementation (i.e., this is just my best first guess) so I think it makes more sense to add tests along with the implementations.

**Review by markmatney — APPROVED** [review]

**Review by DRickard — APPROVED** [review]
